src/containerapp/azext_containerapp/containerapp_decorator.py

```diff
@@ -988,9 +1032,15 @@ def _construct_app_and_env_for_source_or_repo(self):
         env = ContainerAppEnvironment(self.cmd, env_name, env_resource_group, location=location)
         app = ContainerApp(self.cmd, self.get_argument_name(), resource_group, None, image, env, target_port, self.get_argument_registry_server(), self.get_argument_registry_user(), self.get_argument_registry_pass(), self.get_argument_env_vars(), self.get_argument_workload_profile_name(), ingress)
 
-        # Fetch registry credentials
-        _get_registry_details(self.cmd, app, self.get_argument_source())  # fetch ACR creds from arguments registry arguments
+        if self.get_argument_source():
+            _get_registry_details_without_get_creds(self.cmd, app, self.get_argument_source())
+        if self.get_argument_repo():
```

Comment:
what about both --source and --repo provided?

Reply:
We don't allowed input --source and --repo together:
When input --source --repo together, throw:
Cannot use --source and --repo together. Can either deploy from a local directory or a Github repo

For --repo, it will create github action with sourcecontrols, it use the ACR registry to create github action.
[registry_username=app.registry_user,](https://github.com/Azure/azure-cli-extensions/blob/ad5fe47c64e1c874a094db20a7794b283dd2b711/src/containerapp/azext_containerapp/_up_utils.py#L1527)

The sourcecontrols doesn't support registry with registry identity. It only support registry username and password.